### PR TITLE
Update Framework interfaces and mark as API

### LINKED ISSUE #32037: Update Framework interfaces
The following interfaces should be updated in 2.5-develop branch:

1. lib/internal/Magento/Framework/MessageQueue/ConsumerConfigurationInterface.php: Remove deprecated methods
1. lib/internal/Magento/Framework/View/DesignInterface.php: Remove deprecated setArea method
1. lib/internal/Magento/Framework/Webapi/CustomAttributeTypeLocatorInterface.php: Remove deprecated methods

### PR DESCRIPTION
**Pull request by mrtuvn**
### Description (*)

Mark following interfaces as API  + removed deprecated methods usage in code base + added declare strict_types
Update integration tests coverage with changes

lib/internal/Magento/Framework/App/Config/ValueInterface.php @api 
lib/internal/Magento/Framework/MessageQueue/ConsumerConfigurationInterface.php @api + remove deprecated method
lib/internal/Magento/Framework/View/DesignInterface.php mark @api + remove deprecated method setArea from interface 
lib/internal/Magento/Framework/Webapi/CustomAttributeTypeLocatorInterface.php @api + remove deprecated method

### Fixed Issues (if relevant)

1. Fixes magento/magento2#32037

### Manual testing scenarios (*)

1. ...
2. ...

### Questions or comments

CC: @ihor-sviziev @sivaschenko @kandy @pauliarush
### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Theme/Model/View/Design.php
Comment (mrtuvn) [inline app/code/Magento/Theme/Model/View/Design.php]: @sivaschenko @sidolov @gabrieldagama  should we safe for remove this method ? Method defined in interface already removed but i just think we should keep method here for backward compatible

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

File: app/code/Magento/Eav/Model/TypeLocator/SimpleType.php
Comment (mrtuvn) [inline app/code/Magento/Eav/Model/TypeLocator/SimpleType.php:57]: remove public methods in interfaces may violates rule BC principles
https://devdocs.magento.com/contributor-guide/backward-compatible-development/

Comment (sivaschenko) [reply app/code/Magento/Eav/Model/TypeLocator/SimpleType.php:57]: @mrtuvn that's why we are targeting these changes to 2.5-develop, that's the only way to remove deprecated public methods

Comment (mrtuvn) [reply app/code/Magento/Eav/Model/TypeLocator/SimpleType.php:57]: It's safe if i remove all usage of deprecated methods. Ex: setArea i found used in some place

Comment (mrtuvn) [reply app/code/Magento/Eav/Model/TypeLocator/SimpleType.php:57]: removed deprecated mention method

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Integration Tests, Static Tests, Unit Tests, Functional Tests B2B, Functional Tests CE, Functional Tests EE

File: app/code/Magento/Theme/Model/View/Design.php
Comment (ihor-sviziev) [inline app/code/Magento/Theme/Model/View/Design.php:116]: It seems like this property shouldn't be used anymore and should be removed. What do you think?

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

File: app/code/Magento/Theme/Model/View/Design.php
Comment (mrtuvn) [reply app/code/Magento/Theme/Model/View/Design.php:116]: I'm not sure remove that property can make something else break

Comment (ihor-sviziev) [reply app/code/Magento/Theme/Model/View/Design.php:116]: ok. let's leave it as is

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Now we have following error during `bin/magento setup:static-content:deploy -f en_US`:
Error happened during deploy process: Area code is not set


Here is result: https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/32142/47a096751b99127bb2bca87090ccd4a5/Functional/console-error-logs.html

File: app/code/Magento/Theme/Model/View/Design.php
Comment (mrtuvn) [reply app/code/Magento/Theme/Model/View/Design.php:116]: oh no i have removed that @@ Let's see some results tests

**Comment by mrtuvn** [comment]
@sivaschenko any reason why we need remove deprecated method 'setArea' in interface `Magento\Framework\View\DesignInterface`

**Comment by sivaschenko** [comment]
^ @paliarush can you please take a look at this

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
Waiting for answer https://github.com/magento/magento2/pull/32142#issuecomment-780497734 https://github.com/magento/magento2/pull/32142#issuecomment-780662414
For now moving to on hold

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests EE

**Comment by mrtuvn** [comment]
How can i pass Sematic check after removed public methods in interfaces as purpose of this PR ?

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Comment by ihor-sviziev** [comment]
@mrtuvn I believe that's correct behavior, we'll need to get approval for that

File: app/code/Magento/Theme/Model/View/Design.php
Comment (ihor-sviziev) [inline app/code/Magento/Theme/Model/View/Design.php]: could you explain why it's needed? Also not clear how it worked before

Comment (mrtuvn) [reply app/code/Magento/Theme/Model/View/Design.php]: I think we should avoid use `else` in this case -> so go with elseif for check in case `$area` equal null. In case method `setDesignTheme` not pass second argument. Null default value will always be used. So magento will need to call `getArea` instead setArea

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8837](https://jira.corp.magento.com/browse/ENGCOM-8837) has been created to process this Pull Request
:eight_spoked_asterisk: @ihor-sviziev, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by mrtuvn** [comment]
@magento run Semantic Version Checker, Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Comment by mrtuvn** [comment]
Failed B2B related external repo inventory

MC-40489: Billing address fields should be cleared after refreshing checkout page with store pickup delivery method
"Field by name, label, CSS or XPath element with .payment-method._active .billing-address-form input[name='firstname'] was not found."
File related
app/code/Magento/InventoryInStorePickupFrontend/Test/Mftf/Test/StorefrontBillingAddressClearFieldsAfterRefreshingPageTest.xml

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Comment by mrtuvn** [comment]
CC: @engcom-Oscar @kate-kyzyma some test case seem not work after this change

**Comment by mrtuvn** [comment]
This one still same branch 2.5 target because update changes

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run Semantic Version Checker

File: app/code/Magento/Theme/Model/View/Design.php
Comment (Den4ik) [reply app/code/Magento/Theme/Model/View/Design.php]: @mrtuvn What about situation when custom module pass empty string as `$area` for example?

File: app/code/Magento/Theme/Model/View/Design.php
Comment (Den4ik) [reply app/code/Magento/Theme/Model/View/Design.php]: In my opinion we should clear legacy code in major release

**Review by Den4ik — CHANGES_REQUESTED** [review]
@mrtuvn Thanks for work on this PR.
Could you please move all changes for `dev/tests/integration/testsuite/Magento/Theme/Model/View/DesignTest.php` from https://github.com/magento/magento2/pull/32921 into this PR, because as I see you are clearing same files

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@magento run Sample Data Tests EE, Sample Data Tests B2B, Sample Data Tests CE, Integration Tests, Functional Tests EE, Functional Tests B2B, Functional Tests CE

**Comment by Den4ik** [comment]
@magento run Sample Data Tests EE, Sample Data Tests B2B, Sample Data Tests CE, Integration Tests, Functional Tests EE, Functional Tests B2B, Functional Tests CE

**Comment by Den4ik** [comment]
I'm not sure, but something strange on builds of 2.5-develop branch. I see same fails on multiple PR to 2.5. @sivaschenko Could you check please?

**Comment by mrtuvn** [comment]
Maybe this branch out of sync with 2.4. Supprised this changes may break some internal

**Comment by mrtuvn** [comment]
@magento run Integration Tests, Functional Tests EE, Functional Tests B2B, Functional Tests CE

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@gabrieldagama do we have information about tests for 2.5-develop branch? There are many failed tests

**Review by Den4ik — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @Den4ik, thank you for the review.
[ENGCOM-8837](https://jira.corp.magento.com/browse/ENGCOM-8837) has been created to process this Pull Request

File: app/code/Magento/Theme/Model/Theme.php
Comment (mrtuvn) [inline app/code/Magento/Theme/Model/Theme.php:276]: can we refactor this method to simply ? Just call directly return getData('area'), Remove redundance code above

CC: @ihor-sviziev

**Comment by mrtuvn** [comment]
@magento run all tests

File: app/code/Magento/Theme/Model/Theme.php
Comment (tuna2smc) [reply app/code/Magento/Theme/Model/Theme.php:271]: Just curious should use Magento\Framework\App\Area::PARAM_AREA instead string `area`

Comment (mrtuvn) [reply app/code/Magento/Theme/Model/Theme.php:276]: Just curious should use Magento\Framework\App\Area::PARAM_AREA instead string area

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Pull request closed without merging**
